src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/UsageBasedMetadataManager.cs

```diff
@@ -403,6 +403,38 @@ private static bool IsTrimmableAssembly(ModuleDesc assembly)
             return false;
         }
 
+        public override void GetDependenciesDueToEETypePresence(ref DependencyList dependencies, NodeFactory factory, TypeDesc type)
+        {
+            base.GetDependenciesDueToEETypePresence(ref dependencies, factory, type);
+
+            try
+            {
+                if (type.HasBaseType)
+                {
+                    GetDataFlowDependenciesForInstantiation(ref dependencies, factory, type.BaseType, type);
+                }
+
+                foreach (var runtimeInterface in type.RuntimeInterfaces)
+                {
+                    GetDataFlowDependenciesForInstantiation(ref dependencies, factory, runtimeInterface, type);
+                }
```

Comment:
This goes over all interfaces, including those that are inherited from the bases. What's the illinker behavior for.

```
interface IFoo<[DAM(Methods)] T> { }

[Suppress(The warning that Foo's T doesn't have DAM)]
class Foo<T> : IFoo<T> { }

// Does the suppression on Foo apply here? Bar's RuntimeInterfaces includes IFoo
class Bar<T> : Foo<T> { }
```

(This is obviously in the "obscure corner cases very few people will care about" bucket).

Reply:
Good catch - what would be the API to get just the declared interfaces?
It's definitely a corner case, because Roslyn typically flattens out this during compilation anyway, but still.

Reply:
There is `.ExplicitlyImplementedInterfaces` but IIRC because of Roslyn's flattening, `Bar<T>` would still have `IFoo` there even though nobody put it there. We would have to look at interfaces of the base type and whether the problem was introduced in the base type to distinguish it from `class YetAnotherBar<T> : IFoo<T>, Foo<T> { }`. Maybe just leave as wart?

Reply:
I'll look at what linker does (I assume it just reads the IL so will see the same flattening) and probably leave it as is if it does the same.

Reply:
I changed it to go over the explicitly implemented ones only.